src/group_theory/perm/cycles.lean

```diff
@@ -516,6 +516,44 @@ begin
       rw [pow_mul, pow_order_of_eq_one, one_pow] } }
 end
 
+lemma is_cycle.pow_eq_pow_iff [fintype β] [decidable_eq β] {f : perm β} (hf : is_cycle f)
+  {a b : ℕ} : f ^ a = f ^ b ↔ ∃ (x ∈ f.support), (f ^ a) x = (f ^ b) x :=
```

Comment:
Would it be better to avoid `support` here?
```suggestion
  {a b : ℕ} : f ^ a = f ^ b ↔ ∃ x, f x ≠ x ∧ (f ^ a) x = (f ^ b) x :=
```

Reply:
In particular, is the lemma true without the finiteness assumption if phrased this way? (I haven't thought about it)

Reply:
I'd argue it might still might just be more convenient with my phrasing, as my guess is that in most cases it would save the caller a rewrite by mem_support; but happy to let you make the call

Reply:
I only chose this way of stating this because of the prior lemma. If we were to make the change, it should be made to both lemmas. Any thoughts on this @pechersky?

Reply:
Changed both lemmas in the new commit!